### ISSUE TITLE
Barebones Custom Elements API pollyfill

### ISSUE THREAD

**Issue by tbranyen** [tbranyen/diffhtml#16]
To make this library even more useful, it'd be great to provide a Custom Elements pollyfill that can support the following:

``` javascript
var MyElement = Object.create(HTMLElement.prototype, {
  createdCallback: {
    value: function() {
      console.log('The element instance', this);
    }
  },

  attachedCallback: {
    value: function() {
      console.log('The element is now in the DOM');
    }
  },

  detachedCallback: {
    value: function() {
      console.log('The element has left the DOM');
    }
  },

  attributeChangedCallback: {
    value: function(name, previousValue, value) {
      if (previousValue == null) {
        console.log('New attribute with name: ', name, ' and value: ', value);
      } else if (value == null) {
        console.log('Removed attribute with name: ', name,
          ' with the value: ', previousValue);
      } else {
        console.log('Changed attribute with name: ', name,
          ' with the value: ', previousValue, 'to the value: ', value);
      }
    }
  }
});

document.registerElement('my-element', { prototype: MyElement });
```

**Comment by tbranyen** [comment]
For ES6:

``` javascript
class MyElement extends HTMLElement {
  createdCallback() {
    console.log('The element instance', this);
  }

  attachedCallback() {
    console.log('The element is now in the DOM');
  },

  detachedCallback() {
      console.log('The element has left the DOM');
    }
  },

  attributeChangedCallback(name, previousValue, value) {
    if (previousValue == null) {
      console.log('New attribute with name: ', name, ' and value: ', value);
    } else if (value == null) {
      console.log('Removed attribute with name: ', name,
        ' with the value: ', previousValue);
    } else {
      console.log('Changed attribute with name: ', name,
        ' with the value: ', previousValue, 'to the value: ', value);
    }
  }
}

document.registerElement('my-element', MyElement);
```

**Comment by tbranyen** [comment]
For custom registration, do `Object.create(element)` first so that we have the `__proto__` chain set up. Then we can add the callback functions to this instance which is then returned.  Otherwise the browser's implement will kick in.

**Comment by tbranyen** [comment]
Follow work here: https://github.com/tbranyen/diffhtml/pull/28

**Comment by tbranyen** [comment]
Closed via #28
